### PR TITLE
Improve CI debuggability

### PR DESCRIPTION
**Pull request by fcremo**
This PR modifies the CI script in the following ways:
* print the full graph of all the components known to orchestra (not limited to revng-distributable)
* print solved graph for the requested component (for debugging orchestra action scheduler)
* introduce a `$PUSH_CHANGES` environment variable; if not set to 1, the CI script will exit before pushing changes.
  Intended as a failsafe against involuntary pushing changes while debugging the CI.
* print the commit hashes of the binary archives in use

Also:
* adds to `.gitignore` the JSON file where orchestra caches its configuration

### CODE REVIEW

File: .orchestra/ci/ci.sh [hunk 1]
```diff
@@ -148,6 +156,11 @@ for TARGET_COMPONENT in $TARGET_COMPONENTS; do
     fi
 done
 
+if [ "$PUSH_CHANGES" != 1 ]; then
+	echo '$PUSH_CHANGES != 1, exiting without pushing changes'
+	exit $RESULT
+fi
```
Comment (aleclearmind) [inline .orchestra/ci/ci.sh]: This early exit introduces the risk that if we append something at the end of the script it accidentally skipped.
Wrap the rest of the code in the `if`.

File: .orchestra/ci/ci.sh [hunk 2]
```diff
@@ -148,6 +156,11 @@ for TARGET_COMPONENT in $TARGET_COMPONENTS; do
     fi
 done
 
+if [ "$PUSH_CHANGES" != 1 ]; then
```
Comment (aleclearmind) [inline .orchestra/ci/ci.sh]: Please use `test`, for coherence's sake.

**Review by aleclearmind — CHANGES_REQUESTED** [review]

**Comment by aleclearmind** [comment]
This MR blocked by stabilizing the CI hook receiver

**Comment by fcremo** [comment]
@aleclearmind I fixed the issues you pointed out and added the CI hook script

**Comment by aleclearmind** [comment]
Please squash fixup commits in PRs.

Group to be merged (PUSH_CHANGES):
915c6053446ae4384ec9fda54bb1eca5704c27fa
f2bf08f2552f31fdce82a227322650ecfa11b8ee
eebabd821299495b26edde7301528f210e027dc8 (last part)

Group to be merged:
d9853f064e06a95f6767a36ed824ea7521cd223f
68f79270f5ad6b43b34c72cca2a67ce02cc0d85b

Group to be merged (improve CI):
383e10dfacc4764ffd2626f168d42efd42d10682
5313b2a0cd228f178a85c1d05d6c4415addc16f3

I expect the following commits:
* eebabd821299495b26edde7301528f210e027dc8 (first part)
* One for `PUSH_CHANGES`
* One for `ci-hook`
* Improve CI debuggability
* Ignore JSON config cache

**Comment by fcremo** [comment]
Closing this, it has been merged in a1dd992ee58e716973182a4db522a7930ec67004

**Pull request closed without merging**